node/core/dispute-coordinator/src/real/mod.rs

```diff
@@ -499,10 +503,182 @@ async fn handle_new_activations(
 					db::v1::note_current_session(overlay_db, session)?;
 				}
 			},
-			_ => {},
+			Ok(SessionWindowUpdate::Unchanged) => {},
+		};
+		scrape_on_chain_votes(ctx, overlay_db, state, new_leaf, now, metrics).await?;
+	}
+
+	Ok(())
+}
+
+/// Scrapes on-chain votes (backing votes and concluded disputes) for a active leaf of the relay chain.
+async fn scrape_on_chain_votes(
+	ctx: &mut (impl SubsystemContext<Message = DisputeCoordinatorMessage>
+	          + overseer::SubsystemContext<Message = DisputeCoordinatorMessage>),
+	overlay_db: &mut OverlayedBackend<'_, impl Backend>,
+	state: &mut State,
+	new_leaf: Hash,
+	now: u64,
+	metrics: &Metrics,
+) -> Result<(), Error> {
+	// obtain the concluded disputes as well as the candidate backing votes
+	// from the new leaf
+	let ScrapedOnChainVotes { session, backing_validators, disputes } = {
```

Comment:
At the time of block import / inclusion, we don't have any just yet. So that part has to rely on `ApprovalDist` and if we miss those, eventually - given other validators can cast their vote on a dispute - a dispute resolution.

Reply:
Approval votes don't appear on-chain, but we do harvest all off-chain ones already.